modules/backend/formwidgets/RichEditor.php

```diff
@@ -86,6 +87,16 @@ public function prepareVars()
         $this->vars['toolbarButtons'] = $this->evalToolbarButtons();
 
         $this->vars['globalToolbarButtons'] = EditorSetting::getConfigured('html_toolbar_buttons');
+
+        if (!BackendAuth::getUser()->hasPermission(['media.manage_media'])) {
```

Comment:
Avoid using hasPermission, use hasAccess instead as it factors in whether the user is a super user or not. Also, you don't need to pass the check in an array, just a string is fine.